eng/Generate.ps1

```diff
@@ -1,31 +1,22 @@
-param($name, [switch]$noDebug, [switch]$NoReset)
-$ErrorActionPreference = 'Stop'
+#Requires -Version 6.0
+param($name, [switch]$noDebug, [switch]$reset)
 
-function Invoke-Block([scriptblock]$cmd) {
-    $cmd | Out-String | Write-Verbose
-    & $cmd
+$ErrorActionPreference = 'Stop'
 
-    # Need to check both of these cases for errors as they represent different items
-    # - $?: did the powershell script block throw an error
-    # - $lastexitcode: did a windows command executed by the script block end in error
-    if ((-not $?) -or ($lastexitcode -ne 0)) {
-        if ($error -ne $null)
-        {
-            Write-Warning $error[0]
-        }
-        throw "Command failed to execute: $cmd"
+function Invoke-AutoRest($autoRestArguments)
+{
+    $command = "npx autorest-beta $autoRestArguments"
+    Write-Host "> $command"
+    cmd /c "$command 2>&1"
+    if($LastExitCode -ne 0)
+    {
+        Write-Error "Command failed to execute: $command"
     }
 }
 
-function Invoke-AutoRest($autoRestArguments, $repoRoot) {
-    Invoke-Block {
-        $command = "npx autorest-beta $autoRestArguments"
-        $commandText = $command.Replace($repoRoot, "`$(SolutionDir)")
-
-        Write-Host ">" $commandText
-        
-        & cmd /c "$command 2>&1"
-    }
+if ($reset)
```

Comment:
This should be on-by default so CI is always using the correct version

Reply:
How would it not be using the right version? It downloads the version as part of CI, right?

Reply:
autorest.core is cached in user directory and --reset cleans it.

Reply:
Fixed.